homeassistant/core.py

```diff
@@ -576,7 +576,7 @@ def async_create_background_task(
         """
         task = self.loop.create_task(target, name=name)
         self._background_tasks.add(task)
-        task.add_done_callback(self._background_tasks.remove)
+        task.add_done_callback(self._background_tasks.discard)
```

Comment:
I'm not clear how the task would already be removed from `_background_tasks` that would require the need to switch to `discard` as this seems to be the only place that removes it?